### PR TITLE
task 4970858 - Enhance VMInsights plugin logging

### PR DESCRIPTION
**Pull request by mthompson83**
Enhance the plugin logging.

- Adding debug logging in several places.
- Add functionality so that plugin logs always include a specific prefix for ease of identification.
- Added unit tests.

### CODE REVIEW

File: source/code/plugins/VMInsightsDataCollector.rb [hunk 1]
```diff
@@ -99,6 +100,7 @@ def get_filesystems
                             result << Fs.new(a[0], a[6], a[2], a[4]) if a.size == 7
                         rescue ArgumentError => ex
                             # malformed input
+                            @log.debug() { "#{__method__}: #{ex}" }
```
Comment (aydany) [inline source/code/plugins/VMInsightsDataCollector.rb]: Do you want to log the line as well?

File: source/code/plugins/in_vminsights.rb [hunk 2]
```diff
@@ -21,29 +21,30 @@ def configure(conf)
         super
 
         begin
+            @log = LogWrapper.new(LogPrefix, @log)# unless @log.kind_of? LockWrapper
```
Comment (aydany) [inline source/code/plugins/in_vminsights.rb]: Do you want to keep the commented out part?

File: source/code/plugins/VMInsightsDataCollector.rb (hunk 1 above)
Comment (mthompson83) [reply source/code/plugins/VMInsightsDataCollector.rb]: good catch

File: source/code/plugins/in_vminsights.rb (hunk 2 above)
Comment (mthompson83) [reply source/code/plugins/in_vminsights.rb]: Ooops!

**Pull request merged**